packages/mc-scripts/src/bin/cli.ts

```diff
@@ -0,0 +1,244 @@
+import fs from 'fs';
+import path from 'path';
+import { cac } from 'cac';
+import dotenv from 'dotenv';
+import dotenvExpand from 'dotenv-expand';
+import pkgJson from '../../package.json';
+import type {
+  TCliGlobalOptions,
+  TCliCommandBuildOptions,
+  TCliCommandCompileHtmlOptions,
+  TCliCommandConfigSyncOptions,
+} from '../types';
+
+const cli = cac('mc-scripts');
+
+// Makes the script crash on unhandled rejections instead of silently
+// ignoring them. In the future, promise rejections that are not handled will
+// terminate the Node.js process with a non-zero exit code.
+process.on('unhandledRejection', (err) => {
+  throw err;
+});
+
+// Get the current directory where the CLI is executed from. Usually this is the application folder.
+const applicationDirectory = fs.realpathSync(process.cwd());
+
+export const run = () => {
+  cli.option(
+    '--env <path>',
+    `(optional) Parses the file path as a dotenv file and adds the variables to the environment. Multiple flags are allowed.`
+  );
+
+  // Default command
+  cli.command('').action(() => {
+    cli.outputHelp();
+  });
+
+  // Command: start
+  cli
+    .command(
+      'start',
+      'Starts the application in development mode using Webpack Dev Server.'
+    )
+    .alias('dev')
+    .action(async (options: TCliGlobalOptions) => {
+      // Load dotenv files into the process environment.
+      // This is essentially what `dotenv-cli` does, but it's now built into this CLI.
+      loadDotEnvFiles(options);
+
+      // Do this as the first thing so that any code reading it knows the right env.
+      process.env.BABEL_ENV = 'development';
+      process.env.NODE_ENV = 'development';
+
+      const shouldUseExperimentalBundler =
+        process.env.ENABLE_EXPERIMENTAL_VITE_BUNDLER === 'true';
+      if (shouldUseExperimentalBundler) {
+        console.log('Experimental Vite bundler enabled! 🚀');
+        console.log('');
+      }
+
+      const startCommand = shouldUseExperimentalBundler
+        ? await import('../commands/start-vite')
+        : await import('../commands/start');
+      await startCommand.default();
```

Comment:
This is now how we load and execute the commands. The bundle then contains code-splitter chunks.